pkg/storage/plugins/mongodb/plugin.go

```diff
@@ -1,28 +1,38 @@
 package mongodb
 
 import (
+	"fmt"
+
 	"get.porter.sh/porter/pkg/portercontext"
 	"get.porter.sh/porter/pkg/storage/plugins"
+	"get.porter.sh/porter/pkg/storage/pluginstore"
+	"github.com/hashicorp/go-plugin"
 	"github.com/mitchellh/mapstructure"
-	"github.com/pkg/errors"
 )
 
 const PluginKey = plugins.PluginInterface + ".porter.mongodb"
 
+var _ plugins.StorageProtocol = Plugin{}
+
+type Plugin struct {
+	*Store
+}
+
 // PluginConfig are the configuration settings that can be defined for the
 // mongodb plugin in porter.yaml
 type PluginConfig struct {
 	URL     string `mapstructure:"url"`
 	Timeout int    `mapstructure:"timeout,omitempty"`
 }
 
-func NewPlugin(cxt *portercontext.Context, pluginConfig interface{}) (plugins.StoragePlugin, error) {
+func NewPlugin(c *portercontext.Context, rawCfg interface{}) plugin.Plugin {
 	cfg := PluginConfig{
 		Timeout: 10,
 	}
-	if err := mapstructure.Decode(pluginConfig, &cfg); err != nil {
-		return nil, errors.Wrapf(err, "error decoding %s plugin config from %#v", PluginKey, pluginConfig)
+	if err := mapstructure.Decode(rawCfg, &cfg); err != nil {
+		panic(fmt.Errorf("error reading plugin configuration: %w", err))
```

Comment:
The RunInternalPlugin function validates the plugin input before creating the plugins. So there isn't really a path that would lead to this error. I figured I would panic instead of ignoring the error entirely.

There isn't a way to handle the error if we run a plugin and send malformed config on STDIN. I could change the entire call path to bubble up the error, but we are already handling panics in main, so it wouldn't change the behavior anyway?

Reply:
yeah okay this was annoying me too, and this was the push I needed to fix it properly. I've updated the code to return errors